src/System.Private.CoreLib/src/System/Reflection/Assembly.CoreCLR.cs

```diff
@@ -51,7 +51,9 @@ private static Assembly LoadFromResolveHandler(object sender, ResolveEventArgs a
             // in the same folder as it.
             // Form the name of the assembly using the path of the assembly that requested its load.
             AssemblyName requestedAssemblyName = new AssemblyName(args.Name);
-            string requestedAssemblyPath = Path.Combine(Path.GetDirectoryName(requestorPath), requestedAssemblyName.Name+".dll");
+            string requestedAssemblyPath = String.IsNullOrEmpty(requestedAssemblyName.CultureName) ?
+                Path.Combine(Path.GetDirectoryName(requestorPath), requestedAssemblyName.Name+".dll") :
+                Path.Combine(Path.GetDirectoryName(requestorPath), requestedAssemblyName.CultureName, requestedAssemblyName.Name+".dll");
```

Comment:
The test here is pretty simplistic.  It is just looking at an empty assembly name.  If we were concerned we could check the requested assembly.Name ended with `.resources`

Also for culture specific we are also calling `Assembly.LoadFrom`  we could use a different method so that we don't nest these.

Reply:
Please take a look at the same check I implemented for the CDR: https://github.com/vitek-karas/coreclr/blob/309de7b51bef1122a653d4ab73de98bf9f1a7663/src/System.Private.CoreLib/src/System/Runtime/Loader/ComponentDependencyResolver.cs#L82

We need to make sure that the logic is exactly the same as in the native code (ideally we would have a managed version to avoid inconsistency).